### DISCUSSION TITLE
HTTP Authentication

### DISCUSSION THREAD

**Discussion by JeffVM-Solutions** [seleniumbase/SeleniumBase#934] [Q&A] [answered]
Hello Michael, I was wondering if you have any custom settings for HTTP/HTTPS authentication other than passing username and password in the self.open("https://username:password@website.com"). I can imagine you've had to deal with 401 errors while running some scripts.

**Comment by mdmintz** [comment, marked as answer]
Hi @DevOps-Martins , there's another way to handle authentication (via proxy server):
```
pytest --proxy=USERNAME:PASSWORD@IP_ADDRESS:PORT
```
(That ``proxy`` option is unique to SeleniumBase, which essentially lets you run tests through an authenticated Proxy server when using Chrome.)

Additionally, from inside an already running test, you can spin up a new window that uses an authenticated proxy server for a connection:
```
self.get_new_driver(proxy=None)
```
(Replace ``None`` with a proxy string in the format: ``USERNAME:PASSWORD@IP_ADDRESS:PORT``)

**Reply by JeffVM-Solutions** [reply]
Thank you @mdmintz , this issue is more an internal issue with my work environment. We use more than just basic authentication on the web apps and since it's an intranet, proxy server will not be a solution for me.
